presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderWindows.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.WindowNode;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.PriorityQueue;
+
+public class ReorderWindows
```

Comment:
Generally I don't think it is the best approach.
New rules are executed several times, until there is no way that it could rewrite anything. So instead of rewriting everything at once you could just reorder two windows. 

But I am not 100% sure if you should follow this general rule as it could make sorting to be O(n^2). 

Anyway I would not optimize it too early and I try first with a general approach (ordering two windows at once). Once `Memo` get extended to have multiple `PlanNode` per groupId, then sorting with your current approach may (or not, depends on how it be implemented) become more difficult. Lets keep it as simple as possible for now.